en/salesforce/src/docs/TEMPLATE_ms_composer_custom_expression_editor.adoc

```diff
@@ -1,85 +1,304 @@
+
+`addSeconds(2021-03-15T23:45:00,15) = 2021-03-15T23:45:30` +
+
+`addSeconds(2021-03-15T18:23:20, -30) = 2021-03-15T18:22:50`
+|dateTime: DateTime +
+
+numSeconds: number |DateTime
+
+|beginningOfHour |Returns a new DateTime value that changes the Time value in the input to the beginning of the specified hour. The minutes and seconds in the input change to 00:00. |`beginningOfHour(2021-03-15T18:23:20) = 2021-03-15T18:00:00` |dateTime: DateTime |DateTime
+
+|beginningOfDay |Returns a new DateTime value that changes the Time value in the input to the beginning of the specified day. The hours, minutes, and seconds in the input change to 00:00:00. |`beginningOfDay(2021-03-15T18:23:20) = 2021-03-15T00:00:00` |dateTime: DateTime |DateTime
+
+|beginningOfWeek |Returns a new Date value that changes the Day value from the input to the beginning of the first day of the specified week. The function treats Sunday as the first day of the week. |`beginningOfWeek(2021-03-15) = 2021-03-14` +
+
+`beginningOfWeek(today()) = 2021-03-14` |date: Date |Date
+
+|beginningOfWeek |Returns a new DateTime value that changes the Day and Time values from the input to the beginning of the first day of the specified week. The function treats Sunday as the first day of the week.
+|`beginningOfWeek(2021-03-15T18:23:20) = 2021-03-14T00:00:00` +
+
+`beginningOfWeek(now()) = 2021-03-14T00:00:00` |dateTime: DateTime |DateTime
+
+|beginningOfMonth |Returns a new Date value that changes the Day value from the input to the first day of the specified month. |`beginningOfMonth(2021-11-15) = 2021-11-01` +
+
+`beginningOfMonth(today()) = 2021-03-01` |date: Date |Date
+
+|beginningOfMonth |Returns a new DateTime value that changes the Day value from the input to the first day of the specified month. The day and time in the input changes to `01T00:00:00`. |`beginningOfMonth(2021-03-15T18:23:20) = 2021-03-01T00:00:00` +
+
+`beginningOfMonth(now()) = 2021-03-01T00:00:00` |dateTime: DateTime |DateTime
+
+|beginningOfYear |Returns a new Date value that changes the Month and Day values from the input to the first day of the specified year. |`beginningOfYear(2021-11-15) = 2021-01-01` +
+
+`beginningOfYear(today()) = 2021-01-01` |date: Date |Date
+
+|beginningOfYear |Returns a new DateTime value that changes the Month and Day values from the input to the first day of the specified year. It also sets the Time value to `00:00:00`. The month, day, and time in the input changes to `01-01T00:00:00`. |`beginningOfYear(2021-03-15T18:23:20) = 2021-01-01T00:00:00` +
+
+`beginningOfYear(now()) = 2021-01-01T00:00:00` |dateTime: DateTime |DateTime
+
+|daysBetween |Returns the number of days between two dates. If the to Date value is earlier than the from Date value, the function returns a negative number equal to the number of days between the two dates. |`daysBetween(2020-03-15, 2021-03-15) = 365`|from: Date +
```

Comment:
Too bad these aren't ToDate value and FromDate value....

Reply:
Ugh, I know. Developers are silly.